src/server/pfs/server/transaction_access.go

```diff
@@ -0,0 +1,43 @@
+package server
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal"
+	"github.com/pachyderm/pachyderm/v2/src/internal/transactionenv/txncontext"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+type FileAccessor struct {
+	ctx       context.Context
```

Comment:
This seems like it is just moving the problem. Any fileset returned from this shouldn't even be readable with snapshot or serializable

Reply:
I guess I wasn't clear about intention - I planned on explicitly sidestepping that issue for now. This accomplishes the goal of removing other nested transactions and making future ones harder to create. 

This interface makes it easier to implement the compromise you (iirc) mentioned earlier, where filesets are handled by a layer outside of the transaction. We can make the `CreateFileset` method here always return a `TransactionConflict` error itself, and have it add the ID to a list that the txnCtx renews on failure, since any fileset must be created through this object.

Reply:
Talked about this offline. I think the direction here is really cool.